workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/message/SendMessageExecutor.java

```diff
@@ -122,23 +139,30 @@ private List<String> resolveStreamId(ActivityExecutorContext<SendMessage> execut
       return singletonList(event.getStream().getStreamId());
     } else if (execution.getEvent() != null
         && execution.getEvent().getSource() instanceof V4UserJoinedRoom) {
-      // or retrieved from the current even
       V4UserJoinedRoom event = (V4UserJoinedRoom) execution.getEvent().getSource();
       return singletonList(event.getStream().getStreamId());
-    } else if (execution.getEvent() != null
-        && execution.getEvent().getSource() instanceof V4SymphonyElementsAction) {
-      // or retrieved from the current even
-      V4SymphonyElementsAction event = (V4SymphonyElementsAction) execution.getEvent().getSource();
-      return singletonList(event.getStream().getStreamId());
     } else {
       throw new IllegalArgumentException(
           String.format("No stream id set to send a message in activity %s", activity.getId()));
     }
   }
 
-  private String createOrGetStreamId(List<Long> userIds, StreamService streamService) {
-    Stream stream = streamService.create(userIds);
-    return stream.getId();
+  private List<String> createOrGetStreamId(List<Long> userIds, StreamService streamService) {
+    List<String> streamIds = new ArrayList<>();
+
+    for (Long userId : userIds) {
+      // passing a singleton list of long instead of long to make the test mocking easy
+      try {
+        streamIds.add(streamService.create(List.of(userId)).getId());
+      } catch (ApiRuntimeException apiRuntimeException) {
+        // ignore error when user is not found
```

Comment:
@yinan-symphony WDYT? I am not really happy of this handling, do you have a better suggestion?

Reply:
do you mean the ignore on the error when the user is not found? 
yes, it would be great if we can make it traceable, like put the user id in a variable. 
I think we are missing a generic error variable, which holds the potential execution exceptions.
btw, please add a error log for the exception message only, and a debug log for the exception stacktrace.